src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1924,6 +1928,15 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 
 	workerExtendedOpNode = CitusMakeNode(MultiExtendedOp);
 	workerExtendedOpNode->targetList = newTargetEntryList;
+	workerExtendedOpNode->distinctClause = NIL;
```

Comment:
Very very minor. Feel free to ignore.

I would keep `workerExtendedOpNode->groupClauseList = groupClauseList;` just after `workerExtendedOpNode->targetList = newTargetEntryList;`.